### PR TITLE
 Force signedness of shader vertex attributes to match the host.

### PR DESCRIPTION
**Pull request by cdavis5e**
Based on a patch by Stefan Dösinger.

Metal cannot do signedness conversion on vertex attributes, and for good
reason. Putting a `uint4` into an `int4`, or a `char4` into a `uint4`,
would lose those values that are outside the range of the target type.
But putting a `uchar4` into a `short4` or an `int4`, or a `ushort4` into
an `int4`, should work. In that case, force the signedness in the shader
to match the declared type of the host.

### CODE REVIEW

**Comment by cdavis5e** [comment]
Oops, I forgot to update SPIRV-Cross in this change. Stand by...

**Comment by cdavis5e** [comment]
And now, I have another change to SPIRV-Cross that affects this in flight.

File: ExternalRevisions/SPIRV-Cross_repo_revision [hunk 1]
```diff
@@ -1 +1 @@
-ea6bdacd056b77ec1004822482d08b6f31473f25
+52f26ee73648a25e9465035a55b276898f453830
```
[the remark this answers is not in the document]
Comment (billhollings) [reply ExternalRevisions/SPIRV-Cross_repo_revision]: When did you update it...just now? On this PR the change to `SPRIV-Cross_repo_revision` appears to be part of a commit of yours from 6 days ago. Maybe GitHub is grouping them together under the covers then.

[the remark this answers is not in the document]
Comment (billhollings) [reply ExternalRevisions/SPIRV-Cross_repo_revision]: Ah...maybe your `force-push` makes it look that way...and I just picked it up in flight when reviewing.

[the remark this answers is not in the document]
Comment (cdavis5e) [reply ExternalRevisions/SPIRV-Cross_repo_revision]: Well, at least, that's what I usually do when I want to update SPIRV-Cross past some point. It just so happened that my change was HEAD at that point.

If you're looking for a particular commit, you can usually have `git log` grep through the logs for you:

```bash
# Show only changes from a particular author
git log --author='Bill Hollings'
# Show only changes whose log message matches the given regex
git log --grep='My change'
```

`git help log` for more info.

[the remark this answers is not in the document]
Comment (billhollings) [reply ExternalRevisions/SPIRV-Cross_repo_revision]: Yeah that's fine.

It just seemed to me that this MVK PR had prior knowledge of a SPIRV-Cross commit that only came into existence once I merged your SPIRV-Cross PR.

But perhaps you picked it up between the time I merged over at SPRIV-Cross...and updated it here before I got back here to merge your PR's here.

Basically...some form of human-speed race condition. 😉

**Pull request merged**